client/app/adapters/application.js

```diff
@@ -17,4 +32,24 @@ export default class ApplicationAdapter extends JSONAPIAdapter {
 
     return {};
   }
+
+  // Delete the documents array from Snapshot because
+  // package.documents is read-only.
+  // Discarding it from the update prevents
+  // serialization errors that occur when file upload fails.
+  // When file upload fails, recursive ember-file-upload document objects
+  // are inadvertently placed into the package.documents array.
```

Comment:
Actually ember-file-upload documents are placed into the package.documents array by design, by the Attachments component

Reply:
Interesting